ruby/ql/lib/codeql/ruby/ApiGraphs.qll

```diff
@@ -101,9 +101,26 @@ module API {
     DataFlow::ExprNode getAnInstantiation() { result = this.getInstance().getAnImmediateUse() }
 
     /**
-     * Gets a node representing a subclass of the class represented by this node.
+     * Gets a node representing a (direct or indirect) subclass of the class represented by this node.
+     * ```rb
+     * class A; end
+     * class B < A; end
+     * class C < B; end
+     * ```
+     * In the example above, `getMember("A").getASubclass()` will return uses of `A`, `B` and `C`.
      */
-    Node getASubclass() { result = this.getASuccessor(Label::subclass()) }
+    Node getASubclass() { result = this.getAnImmediateSubclass*() }
```

Comment:
Should we update
```ql
Node getInstance() { result = this.getASuccessor(Label::instance()) }
```

to

```ql
Node getInstance() { result = this.getASubclass().getASuccessor(Label::instance()) }
```

@asgerf , @aibaars I think this is related to our discussion yesterday

Reply:
I agree that's probably better (though I still don't have a good intuition about what it's like to model Ruby frameworks).

Reply:
I think that makes sense. If we do so, we should probably also do the same for `getReturn`. Together that should cover both instance and class methods.

Reply:
Added this in https://github.com/github/codeql/pull/7663/commits/704b58519fb1346049ab015c15f6b0618a9e29db